### ISSUE TITLE
Runs on GPU, error on TPU: Computation requires more parameters (546) than supported (limit 236)

### ISSUE THREAD

**Issue by hrbigelow** [pytorch/xla#1963]
## ❓ Questions and Help

Hi all,

Could anyone give a clue what might be going wrong?  I have run this commit, from this [colab](https://colab.research.google.com/drive/1MDY5ZpjlIU9Np4r0xSFNevtGekZT6vZk)

which has produced this output: [debug run](https://drive.google.com/file/d/1-2F1bx1sdjKB1WGDKl2woe_MwZ_0KxOu/view?usp=sharing)

Some lines from it are:

```
Exception in device=TPU:0: Invalid argument: From /job:tpu_worker/replica:0/task:0:
Computation requires more parameters (546) than supported (limit 236).
         [[{{node XRTCompile}}]]
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/torch_xla/distributed/xla_multiprocessing.py", line 119, in _start_fn
    fn(gindex, *args)
  File "ae-wavenet/train.py", line 56, in _mp_fn
    m.train(index)
  File "/content/ae-wavenet/chassis.py", line 127, in train
    loss = self.optim_step_fn()
  File "/content/ae-wavenet/chassis.py", line 95, in <lambda>
    optimizer_args={'closure': self.loss_fn}))
  File "/usr/local/lib/python3.6/dist-packages/torch_xla/core/xla_model.py", line 538, in optimizer_step
    loss = optimizer.step(**optimizer_args)
  File "/usr/local/lib/python3.6/dist-packages/torch/autograd/grad_mode.py", line 15, in decorate_context
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/torch/optim/adam.py", line 62, in step
    loss = closure()
  File "/content/ae-wavenet/chassis.py", line 178, in loss_fn
    self.run_batch()
  File "/content/ae-wavenet/chassis.py", line 170, in run_batch
    batch = next(self.data_iter)
  File "/content/ae-wavenet/chassis.py", line 34, in __next__
    vb = self.per_dev_loader.__next__()[0]
  File "/usr/local/lib/python3.6/dist-packages/torch_xla/distributed/parallel_loader.py", line 31, in __next__
    return self.next()
  File "/usr/local/lib/python3.6/dist-packages/torch_xla/distributed/parallel_loader.py", line 34, in next
    xm.mark_step()
  File "/usr/local/lib/python3.6/dist-packages/torch_xla/core/xla_model.py", line 477, in mark_step
    wait=xu.getenv_as('XLA_SYNC_WAIT', bool, False))
RuntimeError: Invalid argument: From /job:tpu_worker/replica:0/task:0:
Computation requires more parameters (546) than supported (limit 236).
         [[{{node XRTCompile}}]]
Writing run results to /tmp/debug_run-eef90b0a0f8e-root-0
XLA Environment:
  XRT_TPU_CONFIG=tpu_worker;0;10.74.90.234:8470
  TF_FORCE_GPU_ALLOW_GROWTH=true
  XLA_IR_DEBUG=1
  XLA_HLO_DEBUG=1
  TF_CPP_LOG_THREAD_ID=1
  TF_CPP_VMODULE=tensor=5,computation_client=5,xrt_computation_client=5,aten_xla_type=1
  XLA_SAVE_TENSORS_FILE=/tmp/debug_run-eef90b0a0f8e-root-0/graphs
  XLA_METRICS_FILE=/tmp/debug_run-eef90b0a0f8e-root-0/metrics
```

The same code has run successfully on my GTX1070 Max-Q laptop environment with PyTorch version 1.3.1

I've never seen the error before (but it has been several months since I've used torch_xla)

Thanks in advance!

**Comment by dlibenzi** [comment]
@jysohn23 this stuff keeps cropping up. Seems like continuations are enabled in the next-gen executor.
Did we find the clue?

**Comment by jysohn23** [comment]
Hey @hrbigelow I don't have access to the datasets you're using on your colab gdrive, but it looks like you're using an old colab template. Use this in your first setup cell instead, which should lead you to our new runtime:

```
VERSION = "20200325"  #@param ["1.5" , "20200325", "nightly"]
!curl https://raw.githubusercontent.com/pytorch/xla/master/contrib/scripts/env-setup.py -o pytorch-xla-env-setup.py
!python pytorch-xla-env-setup.py --version $VERSION
```

And yes @dlibenzi, similar to this case the command to update the runtime was targeting old runtime but as long as we use this setup script things it should correctly update to new runtime.

**Comment by hrbigelow** [comment]
Hi @jysohn23 @dlibenzi 

Thanks very much - it works with that new preamble.

So by the way for future reference, if I wanted to make the colab runnable for you, what else would I need to do?  One thing is, I have to re-mount my gdrive each time I reconnect, so I'm not sure if that part would be reproducible for you.  Is there a better place to host and store data files for use with Colab, so that I can allow others to run it?

Thanks again,

Henry

**Comment by jysohn23** [comment]
I think if you could reproduce the error with some fake data generator that'd be ideal, if not putting it up temporarily somewhere in like GCS bucket would work for us too. Davide may have some other opinions.

**Comment by hrbigelow** [comment]
Ahh good idea.  And thanks for the sleek preamble, much cleaner.

**Comment by vihari** [comment]
I was running on GCP with Jupyter Notebook and faced the exact problem.  
In my case, it turns it is because I did not set the TPU software version to pytorch-1.6. The mismatch in my conda env and TPU version caused this cryptic error.

**Comment by qshzh** [comment]
I encountered the similar problem when running a simple cifar classification task, it raises the error after about 2000 iterations
```shell
Exception in device=TPU:7: Invalid argument: From /job:tpu_worker/replica:0/task:0:
2 root error(s) found.
  (0) Invalid argument: Computation requires more parameters (6096) than supported (limit 3306).
	 [[{{node XRTCompile}}]]
  (1) Invalid argument: Computation requires more parameters (6096) than supported (limit 3306).
	 [[{{node XRTCompile}}]]
	 [[XRTCompile_G3]]
0 successful operations.
0 derived errors ignored.
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/torch_xla/distributed/xla_multiprocessing.py", line 329, in _mp_start_fn
    _start_fn(index, pf_cfg, fn, args)
  File "/usr/local/lib/python3.7/dist-packages/torch_xla/distributed/xla_multiprocessing.py", line 323, in _start_fn
    fn(gindex, *args)
  File "/content/jam/jamtorch/xla/utils.py", line 78, in new_fn
    value = func(config)
  File "/content/jam/example/jamtorch/tpuddp/main.py", line 32, in run
    trainer.train()
  File "/content/jam/jamtorch/trainer/genetic_trainer.py", line 212, in train
    self.train_step(batch)
  File "/content/jam/jamtorch/trainer/genetic_trainer.py", line 240, in train_step
    if self.loss_backward(loss):
  File "/content/jam/jamtorch/trainer/genetic_trainer.py", line 258, in loss_backward
    loss.backward()
  File "/usr/local/lib/python3.7/dist-packages/torch/tensor.py", line 245, in backward
    torch.autograd.backward(self, gradient, retain_graph, create_graph, inputs=inputs)
  File "/usr/local/lib/python3.7/dist-packages/torch/autograd/__init__.py", line 147, in backward
    allow_unreachable=True, accumulate_grad=True)  # allow_unreachable flag
RuntimeError: Invalid argument: From /job:tpu_worker/replica:0/task:0:
2 root error(s) found.
  (0) Invalid argument: Computation requires more parameters (6096) than supported (limit 3306).
	 [[{{node XRTCompile}}]]
  (1) Invalid argument: Computation requires more parameters (6096) than supported (limit 3306).
	 [[{{node XRTCompile}}]]
	 [[XRTCompile_G3]]
0 successful operations.
0 derived errors ignored.
```

**Comment by JackCaoG** [comment]
Yea, it is the hard limit of number of parameter right now. I will work on a change on the pt/xla side to pass the parameter as a tuple and that should solve this error.

**Comment by rwightman** [comment]
@JackCaoG I just ran into this error and same limit as last poster `(limit 3306)` ... working with a larger model (but not unreasonably large) that should fit in memory fine... I was actually training it with the TPUv4 while I had alpha access and it was going well, cannot get it working with a v3 TPU-VM. Using PyTorch XLA 1.10

**Comment by JackCaoG** [comment]
I dropped this project to work on something else last year. @rwightman My guess is that v4 we have larger `smem` hence more parameter can fit in. We are working on runtime migration which will solve this issue permanently. The target timeline is end of June. We can potentially fix this issue on existing runtime(my estimation is it will take ~ 2 weeks) but it will likely be wasted in ~ 6 months when we do the switch. How urgent is your use case?

**Comment by rwightman** [comment]
@JackCaoG I'm running through some larger candidate vision models for medium-large scale CLIP / LiT / etc image-text model pretraining. I hope to include a script with working hparams for reproducing such training on TPU, GPU, (maybe IPU) w/ PyTorch ... so models are fairly large, and hope to go a bit larger still... but so far I've kept within what I thought would be reasonable to test on a single v3-8. I can resume training this one on a 4x GPU machine so no urgency there. 

Once I sort out the rest of the setup and get further along with the runs on larger dataset will likely run into this limit. Not sure how long all that will take but I can probably work around this for a bit. It does appear that it'd be easy to hit in any scenario where pod use is needed (models too large to fit decent batch sizes on a single accelerator). So surprised more people haven't hit it.

**Comment by JackCaoG** [comment]
sounds good, I will keep you updated regarding this issue.

**Comment by ikergarcia1996** [comment]
Same problem here, I can reproduce the error even with a tiny 9M parameter model. 
TPU v3-8
tensorflow 2.9.0
torch 1.11.0
torch-xla 1.11
pytorch-lightning==1.6.0rc1
cloud-tpu-client==0.10

I am trying to train this model: https://github.com/ikergarcia1996/Self-Driving-Car-in-Video-Games/blob/master/model.py#L778 from this repository: https://github.com/ikergarcia1996/Self-Driving-Car-in-Video-Games

@JackCaoG  Any idea what could be causing this error even with such small models? It would be great to be able to take advantage of the 31 free days of the TRC program to train a model.

```

  | Name                         | Type                | Params 
----------------------------------------------------------------------
0  | model                        | TEDD1104Transformer | 9.3 M  
1  | train_accuracy               | Accuracy            | 0
2  | test_accuracy_k1_macro       | Accuracy            | 0
3  | test_accuracy_k3_micro       | Accuracy            | 0
4  | validation_accuracy_k1_micro | Accuracy            | 0
5  | validation_accuracy_k3_micro | Accuracy            | 0
6  | validation_accuracy_k1_macro | Accuracy            | 0
7  | validation_accuracy_k3_macro | Accuracy            | 0
8  | test_accuracy_k1_micro       | Accuracy            | 0
9  | test_accuracy_k3_macro       | Accuracy            | 0
10 | validation_distance          | MeanSquaredError    | 0
11 | criterion                    | WeightedMseLoss     | 0
12 | Controller2Keyboard          | Controller2Keyboard | 0
----------------------------------------------------------------------
9.3 M     Trainable params
0         Non-trainable params
9.3 M     Total params
37.374    Total estimated model params size (MB)
Epoch 0:   0%|                                                                                                                   | 8/39963 [04:34<380:50:43, 34.31s/it, loss=0.43, v_num=base]
2022-03-28 11:57:16.078552: E tensorflow/core/tpu/kernels/tpu_compilation_cache_external.cc:113] Computation requires more parameters (3311) than supported (limit 3305).
2022-03-28 11:57:16.078644: F tensorflow/core/tpu/kernels/tpu_program_group.cc:86] Check failed: xla_tpu_programs.size() > 0 (0 vs. 0)
https://symbolize.stripped_domain/r/?trace=7f19fc95103b,7f19fc9510bf,7f18fed31bcf,7f18f93a6922,7f18f9364ebd,7f18f93b4db0,7f18f93b48ae,7f18f5216ed3,7f18fa8581b8,7f18fe7e38a0,7f18fe7e5633,7f18
fecfacb1,7f18fecfa4e0,7f18fece28cb,7f19fc8f3608&map=b5462df73b9bb298b2bca5d2f02176eed80a2e90:7f18f08d2000-7f1901bc7e30
*** SIGABRT received by PID 714187 (TID 714993) on cpu 24 from PID 714187; stack trace: ***
PC: @     0x7f19fc95103b  (unknown)  raise
    @     0x7f18efd34cda        992  (unknown)
    @     0x7f19fc9510c0       3968  (unknown)
    @     0x7f18fed31bd0         16  tensorflow::internal::LogMessageFatal::~LogMessageFatal()
    @     0x7f18f93a6923        592  tensorflow::tpu::TpuProgramGroup::Initialize()
    @     0x7f18f9364ebe       1488  tensorflow::tpu::TpuCompilationCacheExternal::InitializeEntry()
    @     0x7f18f93b4db1        800  tensorflow::tpu::TpuCompilationCacheInterface::CompileIfKeyAbsentHelper()
    @     0x7f18f93b48af        496  tensorflow::tpu::TpuCompilationCacheInterface::CompileIfKeyAbsent()
    @     0x7f18f5216ed4        912  tensorflow::XRTCompileOp::Compute()
    @     0x7f18fa8581b9        432  tensorflow::XlaDevice::Compute()
    @     0x7f18fe7e38a1       2128  tensorflow::(anonymous namespace)::ExecutorState<>::Process()
    @     0x7f18fe7e5634         48  std::_Function_handler<>::_M_invoke()
    @     0x7f18fecfacb2        128  Eigen::ThreadPoolTempl<>::WorkerLoop()
    @     0x7f18fecfa4e1         48  tensorflow::thread::EigenEnvironment::CreateThread()::{lambda()#1}::operator()()
    @     0x7f18fece28cc         80  tensorflow::(anonymous namespace)::PThread::ThreadFn()
    @     0x7f19fc8f3609  (unknown)  start_thread
https://symbolize.stripped_domain/r/?trace=7f19fc95103b,7f18efd34cd9,7f19fc9510bf,7f18fed31bcf,7f18f93a6922,7f18f9364ebd,7f18f93b4db0,7f18f93b48ae,7f18f5216ed3,7f18fa8581b8,7f18fe7e38a0,7f18
fe7e5633,7f18fecfacb1,7f18fecfa4e0,7f18fece28cb,7f19fc8f3608&map=b5462df73b9bb298b2bca5d2f02176eed80a2e90:7f18f08d2000-7f1901bc7e30,50c831e765011c7eb7163b7f3cb5c4b6:7f18e158a000-7f18f00a2f00

E0328 11:57:16.307495  714993 coredump_hook.cc:365] RAW: Remote crash data gathering hook invoked.
E0328 11:57:16.307515  714993 coredump_hook.cc:411] RAW: Skipping coredump since rlimit was 0 at process start.
E0328 11:57:16.307529  714993 client.cc:222] RAW: Coroner client retries enabled (b/136286901), will retry for up to 30 sec.
E0328 11:57:16.307539  714993 coredump_hook.cc:473] RAW: Sending fingerprint to remote end.
E0328 11:57:16.307550  714993 coredump_socket.cc:124] RAW: Stat failed errno=2 on socket /var/google/services/logmanagerd/remote_coredump.socket
E0328 11:57:16.307559  714993 coredump_hook.cc:477] RAW: Cannot send fingerprint to Coroner: [NOT_FOUND] Missing crash reporting socket. Is the listener running?
E0328 11:57:16.307565  714993 coredump_hook.cc:550] RAW: Discarding core.
E0328 11:57:16.732287  714993 process_state.cc:771] RAW: Raising signal 6 with default behavior
2022-03-28 11:57:17.327159: W tensorflow/core/distributed_runtime/rpc/grpc_remote_master.cc:157] RPC failed with status = "UNAVAILABLE: Socket closed" and grpc_error_string = "{"created":"@1648468637.326935100","description":"Error received from peer ipv4:127.0.0.1:51011","file":"external/com_github_grpc_grpc/src/core/lib/surface/call.cc","file_line":1056,"grpc_message":"Socket closed","grpc_status":14}", maybe retrying the RPC
2022-03-28 11:57:22.529565: W tensorflow/core/distributed_runtime/rpc/grpc_remote_master.cc:157] RPC failed with status = "UNAVAILABLE: Socket closed" and grpc_error_string = "{"created":"@1648468642.529380655","description":"Error received from peer ipv4:127.0.0.1:51011","file":"external/com_github_grpc_grpc/src/core/lib/surface/call.cc","file_line":1056,"grpc_message":"Socket closed","grpc_status":14}", maybe retrying the RPC
2022-03-28 11:57:22.529565: W tensorflow/core/distributed_runtime/rpc/grpc_remote_master.cc:157] RPC failed with status = "UNAVAILABLE: Socket closed" and grpc_error_string = "{"created":"@1648468642.529380655","description":"Error received from peer ipv4:127.0.0.1:51011","file":"external/com_github_grpc_grpc/src/core/lib/surface/call.cc","file_line":1056,"grpc_message":"Socket closed","grpc_status":14}", maybe retrying the RPC
2022-03-28 11:57:22.529851: W tensorflow/core/distributed_runtime/rpc/grpc_remote_master.cc:157] RPC failed with status = "UNAVAILABLE: Socket closed" and grpc_error_string = "{"created":"@1648468642.529622009","description":"Error received from peer ipv4:127.0.0.1:51011","file":"external/com_github_grpc_grpc/src/core/lib/surface/call.cc","file_line":1056,"grpc_message":"Socket closed","grpc_status":14}", maybe retrying the RPC
2022-03-28 11:57:22.530091: W tensorflow/core/distributed_runtime/rpc/grpc_remote_master.cc:157] RPC failed with status = "UNAVAILABLE: Socket closed" and grpc_error_string = "{"created":"@1648468642.529921516","description":"Error received from peer ipv4:127.0.0.1:51011","file":"external/com_github_grpc_grpc/src/core/lib/surface/call.cc","file_line":1056,"grpc_message":"Socket closed","grpc_status":14}", maybe retrying the RPC
2022-03-28 11:57:22.574131: W tensorflow/core/distributed_runtime/rpc/grpc_remote_master.cc:157] RPC failed with status = "UNAVAILABLE: Socket closed" and grpc_error_string = "{"created":"@1648468642.573979797","description":"Error received from peer ipv4:127.0.0.1:51011","file":"external/com_github_grpc_grpc/src/core/lib/surface/call.cc","file_line":1056,"grpc_message":"Socket closed","grpc_status":14}", maybe retrying the RPC
2022-03-28 11:57:22.574730: W tensorflow/core/distributed_runtime/rpc/grpc_remote_master.cc:157] RPC failed with status = "UNAVAILABLE: Connection reset by peer" and grpc_error_string = "{"created":"@1648468642.574629478","description":"Error received from peer ipv4:127.0.0.1:51011","file":"external/com_github_grpc_grpc/src/core/lib/surface/call.cc","file_line":1056,"grpc_message":"Connection reset by peer","grpc_status":14}", maybe retrying the RPC
2022-03-28 11:57:22.574755: W tensorflow/core/distributed_runtime/rpc/grpc_remote_master.cc:157] RPC failed with status = "UNAVAILABLE: Connection reset by peer" and grpc_error_string = "{"created":"@1648468642.574701121","description":"Error received from peer ipv4:127.0.0.1:51011","file":"external/com_github_grpc_grpc/src/core/lib/surface/call.cc","file_line":1056,"grpc_message":"Connection reset by peer","grpc_status":14}", maybe retrying the RPC
2022-03-28 11:57:22.574797: W tensorflow/core/distributed_runtime/rpc/grpc_remote_master.cc:157] RPC failed with status = "UNAVAILABLE: Connection reset by peer" and grpc_error_string = "{"created":"@1648468642.574746230","description":"Error received from peer ipv4:127.0.0.1:51011","file":"external/com_github_grpc_grpc/src/core/lib/surface/call.cc","file_line":1056,"grpc_message":"Connection reset by peer","grpc_status":14}", maybe retrying the RPC
2022-03-28 11:57:22.574778: W tensorflow/core/distributed_runtime/rpc/grpc_remote_master.cc:157] RPC failed with status = "UNAVAILABLE: Socket closed" and grpc_error_string = "{"created":"@1648468642.574625986","description":"Error received from peer ipv4:127.0.0.1:51011","file":"external/com_github_grpc_grpc/src/core/lib/surface/call.cc","file_line":1056,"grpc_message":"Socket closed","grpc_status":14}", maybe retrying the RPC
2022-03-28 11:57:22.574723: W tensorflow/core/distributed_runtime/rpc/grpc_remote_master.cc:157] RPC failed with status = "UNAVAILABLE: Socket closed" and grpc_error_string = "{"created":"@1648468642.574472508","description":"Error received from peer ipv4:127.0.0.1:51011","file":"external/com_github_grpc_grpc/src/core/lib/surface/call.cc","file_line":1056,"grpc_message":"Socket closed","grpc_status":14}", maybe retrying the RPC
2022-03-28 11:57:22.574892: W tensorflow/core/distributed_runtime/rpc/grpc_remote_master.cc:157] RPC failed with status = "UNAVAILABLE: Connection reset by peer" and grpc_error_string = "{"created":"@1648468642.574855850","description":"Error received from peer ipv4:127.0.0.1:51011","file":"external/com_github_grpc_grpc/src/core/lib/surface/call.cc","file_line":1056,"grpc_message":"Connection reset by peer","grpc_status":14}", maybe retrying the RPC
https://symbolize.stripped_domain/r/?trace=7f19fc8fa376,7f19fc9510bf,0&map=
*** SIGTERM received by PID 714282 (TID 714282) on cpu 63 from PID 711980; stack trace: ***
PC: @     0x7f19fc8fa376  (unknown)  pthread_cond_wait@@GLIBC_2.3.2
    @     0x7f18efd34cda        992  (unknown)
    @     0x7f19fc9510c0  (unknown)  (unknown)
    @                0x1  (unknown)  (unknown)
https://symbolize.stripped_domain/r/?trace=7f19fc8fa376,7f18efd34cd9,7f19fc9510bf,0&map=50c831e765011c7eb7163b7f3cb5c4b6:7f18e158a000-7f18f00a2f00
E0328 11:57:24.459454  714282 coredump_hook.cc:320] RAW: Remote crash gathering disabled for SIGTERM.
E0328 11:57:24.487866  714282 process_state.cc:771] RAW: Raising signal 15 with default behavior
https://symbolize.stripped_domain/r/?trace=7f19fc8fa376,7f19fc9510bf,0&map=
*** SIGTERM received by PID 714288 (TID 714288) on cpu 72 from PID 711980; stack trace: ***
PC: @     0x7f19fc8fa376  (unknown)  pthread_cond_wait@@GLIBC_2.3.2
    @     0x7f18efd34cda        992  (unknown)
    @     0x7f19fc9510c0  (unknown)  (unknown)
    @                0x1  (unknown)  (unknown)
https://symbolize.stripped_domain/r/?trace=7f19fc8fa376,7f18efd34cd9,7f19fc9510bf,0&map=50c831e765011c7eb7163b7f3cb5c4b6:7f18e158a000-7f18f00a2f00
E0328 11:57:24.659237  714288 coredump_hook.cc:320] RAW: Remote crash gathering disabled for SIGTERM.
E0328 11:57:24.687005  714288 process_state.cc:771] RAW: Raising signal 15 with default behavior
https://symbolize.stripped_domain/r/?trace=7f19fc8fa376,7f19fc9510bf,0&map=
*** SIGTERM received by PID 714292 (TID 714292) on cpu 80 from PID 711980; stack trace: ***
PC: @     0x7f19fc8fa376  (unknown)  pthread_cond_wait@@GLIBC_2.3.2
    @     0x7f18efd34cda        992  (unknown)
    @     0x7f19fc9510c0  (unknown)  (unknown)
    @                0x1  (unknown)  (unknown)
https://symbolize.stripped_domain/r/?trace=7f19fc8fa376,7f18efd34cd9,7f19fc9510bf,0&map=50c831e765011c7eb7163b7f3cb5c4b6:7f18e158a000-7f18f00a2f00
E0328 11:57:24.788845  714292 coredump_hook.cc:320] RAW: Remote crash gathering disabled for SIGTERM.
E0328 11:57:24.817159  714292 process_state.cc:771] RAW: Raising signal 15 with default behavior
https://symbolize.stripped_domain/r/?trace=7f19fc8fa376,7f19fc9510bf,0&map=
*** SIGTERM received by PID 714296 (TID 714296) on cpu 63 from PID 711980; stack trace: ***
PC: @     0x7f19fc8fa376  (unknown)  pthread_cond_wait@@GLIBC_2.3.2
    @     0x7f18efd34cda        992  (unknown)
    @     0x7f19fc9510c0  (unknown)  (unknown)
    @                0x1  (unknown)  (unknown)

...

Traceback (most recent call last):
  File "train.py", line 655, in <module>
    train_new_model(
  File "train.py", line 238, in train_new_model
    train(
  File "train.py", line 107, in train
    trainer.fit(model, datamodule=data)
  File "/home/ikergarcia/.local/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 773, in fit
    self._call_and_handle_interrupt(
  File "/home/ikergarcia/.local/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 724, in _call_and_handle_interrupt
    return self.strategy.launcher.launch(trainer_fn, *args, trainer=self, **kwargs)
  File "/home/ikergarcia/.local/lib/python3.8/site-packages/pytorch_lightning/strategies/launchers/xla_spawn.py", line 76, in launch
    xmp.spawn(
  File "/usr/local/lib/python3.8/dist-packages/torch_xla/distributed/xla_multiprocessing.py", line 389, in spawn
    return torch.multiprocessing.start_processes(
  File "/usr/local/lib/python3.8/dist-packages/torch/multiprocessing/spawn.py", line 198, in start_processes
    while not context.join():
  File "/usr/local/lib/python3.8/dist-packages/torch/multiprocessing/spawn.py", line 140, in join
    raise ProcessExitedException(
torch.multiprocessing.spawn.ProcessExitedException: process 0 terminated with signal SIGABRT
Traceback (most recent call last):
  File "train.py", line 655, in <module>
    train_new_model(
  File "train.py", line 238, in train_new_model
    train(
  File "train.py", line 107, in train
    trainer.fit(model, datamodule=data)
  File "/home/ikergarcia/.local/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 773, in fit
    self._call_and_handle_interrupt(
  File "/home/ikergarcia/.local/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 724, in _call_and_handle_interrupt
    return self.strategy.launcher.launch(trainer_fn, *args, trainer=self, **kwargs)
  File "/home/ikergarcia/.local/lib/python3.8/site-packages/pytorch_lightning/strategies/launchers/xla_spawn.py", line 76, in launch
    xmp.spawn(
  File "/usr/local/lib/python3.8/dist-packages/torch_xla/distributed/xla_multiprocessing.py", line 389, in spawn
    return torch.multiprocessing.start_processes(
  File "/usr/local/lib/python3.8/dist-packages/torch/multiprocessing/spawn.py", line 198, in start_processes
    while not context.join():
  File "/usr/local/lib/python3.8/dist-packages/torch/multiprocessing/spawn.py", line 140, in join
    raise ProcessExitedException(
torch.multiprocessing.spawn.ProcessExitedException: process 0 terminated with signal SIGABRT
Traceback (most recent call last):
  File "train.py", line 655, in <module>
    train_new_model(
  File "train.py", line 238, in train_new_model
train(
  File "train.py", line 107, in train
    trainer.fit(model, datamodule=data)
  File "/home/ikergarcia/.local/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 773, in fit
    self._call_and_handle_interrupt(
  File "/home/ikergarcia/.local/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 724, in _call_and_handle_interrupt
    return self.strategy.launcher.launch(trainer_fn, *args, trainer=self, **kwargs)
  File "/home/ikergarcia/.local/lib/python3.8/site-packages/pytorch_lightning/strategies/launchers/xla_spawn.py", line 76, in launch
    xmp.spawn(
  File "/usr/local/lib/python3.8/dist-packages/torch_xla/distributed/xla_multiprocessing.py", line 389, in spawn
    return torch.multiprocessing.start_processes(
  File "/usr/local/lib/python3.8/dist-packages/torch/multiprocessing/spawn.py", line 198, in start_processes
    while not context.join():
  File "/usr/local/lib/python3.8/dist-packages/torch/multiprocessing/spawn.py", line 140, in join
    raise ProcessExitedException(
torch.multiprocessing.spawn.ProcessExitedException: process 0 terminated with signal SIGABRT



```
